qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/presenter.py

```diff
@@ -0,0 +1,91 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+# pylint: disable=invalid-name
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.tabs.common import INSTRUMENT_DICT
+from mantidqt.utils.observer_pattern import GenericObserverWithArgPassing
+
+
+class GSAS2Presenter(object):
+    def __init__(self, model, view):
+        self.model = model
+        self.view = view
+
+        self.rb_num = None
+        self.instrument = "ENGINX"
+        self.current_plot_index = None
+        self.focus_run_observer_gsas2 = GenericObserverWithArgPassing(
+            self.view.set_default_gss_files)
+        self.prm_filepath_observer_gsas2 = GenericObserverWithArgPassing(
+            self.view.set_default_prm_files)
+        self.connect_view_signals()
+
+        self.latest_load_parameters = None
+
+    def connect_view_signals(self):
+        self.view.set_refine_clicked(self.on_refine_clicked)
+        self.view.number_output_histograms_combobox.currentTextChanged.connect(self.on_plot_index_changed)
+
+    def on_refine_clicked(self):
+        self.clear_plot()
+        load_params = self.view.get_load_parameters()
+        project_name = self.view.get_project_name()
+        refine_params = self.view.get_refinement_parameters()
+        number_output_histograms = self.model.run_model(load_params, refine_params, project_name,
+                                                        self.rb_num, self.get_limits_if_same_load_parameters())
+        self.plot_result(1)
+        self.view.set_number_histograms(number_output_histograms)
+        self.save_latest_load_parameters()
+
+    def save_latest_load_parameters(self):
+        self.latest_load_parameters = self.view.get_load_parameters()
+
+    def get_limits_if_same_load_parameters(self):
+        new_load_params = self.view.get_load_parameters()
+        if not new_load_params:
+            return None
```

Comment:
I tried hitting the "Refine in GSAS II" button with an invalid Focused Data file and I got an error "Could not determine Minimum and Maximum X values ...". I wonder if this check here should have caught this? The `new_load_params` variable was a list containing 3 items and the 3rd one was empty

Reply:
I have caught this in the model instead by checking that each of the filepath parameters are not None